lambdas/es/indexer/index.py

```diff
@@ -108,8 +117,107 @@ def infer_extensions(key, ext):
     return ext
 
 
-def get_contents(bucket, key, ext, *, etag, version_id, s3_client, size):
-    """get the byte contents of a file"""
+@retry(
+    stop=stop_after_attempt(MAX_RETRY),
+    wait=wait_exponential(multiplier=2, min=4, max=30),
+    retry=(retry_if_exception(should_retry_exception))
+)
+def select_manifest_meta(s3_client, bucket: str, key: str):
+    """use s3 select to quickly extract line 1 of manifest
+        returns:
+            * first line (if found), None if not found or error
+    """
+    try:
+        raw = s3_client.select_object_content(
+            Bucket=bucket,
+            Key=key,
+            Expression=SELECT_PACKAGE_META,
+            ExpressionType="SQL",
+            InputSerialization={"JSON": {"Type": "LINES"}},
+            OutputSerialization={"JSON": {}}
+        )
+        for event in raw["Payload"]:
+            if "Records" in event:
+                data = event["Records"]["Payload"]
+                return data
+    except botocore.exceptions.ClientError as cle:
+        print(f"Unable to S3 select manifest: {cle}")
+
+    return None
+
+
+def index_if_manifest(
+        s3_client,
+        doc_queue: DocumentQueue,
+        event_type: str,
+        *,
+        bucket: str,
+        etag: str,
+        ext: str,
+        key: str,
+        last_modified: str,
+        version_id: Optional[str],
+        size: int
+) -> bool:
+    """index manifest files as package documents in ES
+        Returns:
+            - True if manifest (and passes to doc_queue for indexing)
+            - False if not a manifest (no attempt at indexing)
+    """
+    pointer_prefix, pointer_file = split(key)
+    if not pointer_prefix.startswith(POINTER_PREFIX_V1):
+        return False
+    try:
+        manifest_timestamp = int(pointer_file)
+        if (
+                manifest_timestamp < 1451631600  # 1/1/2016 (oldest manifest)
+                or manifest_timestamp > 1767250800  # 1/1/2026 (no V1 anymore)
+        ):
+            raise ValueError(f"Invalid manifest pointer s3://{bucket}{key}")
+        package_hash = get_plain_text(
+            bucket,
+            key,
+            size,
+            None,
+            etag=etag,
+            s3_client=s3_client,
+            version_id=version_id
+        ).strip()
+    except ValueError as err:
+        print(f"Unexpected manifest pointer file: s3://{bucket}/{key}: {err}")
+        return False
```

Comment:
Consider rewriting like this:
```suggestion
    err_msg = None
    try:
        manifest_timestamp = int(pointer_file)
    except ValueError as err:
        err_msg = f"Unexpected manifest pointer file: s3://{bucket}/{key}: {err}"
    else:
        if not 1451631600 <= manifest_timestamp <= 1767250800:
            err_msg = f"Invalid manifest pointer s3://{bucket}{key}"
    if err_msg:
        print(err_msg)
        return False

    package_hash = get_plain_text(
        bucket,
        key,
        size,
        None,
        etag=etag,
        s3_client=s3_client,
        version_id=version_id,
    ).strip()
```